### ISSUE TITLE
movement:get_type() or entity:get_movement_type()

### ISSUE THREAD

**Issue by wrightmat** [christopho/solarus#840]
Unless I've missed something in the API, it is currently possible to retrieve an entity's movement, but not to obtain the type of movement that is. As there are several other functions to obtain properties of movements, movement:get_type would probably make the most sense of the options.

**Comment by Diarandor** [comment]
I completely agree that movement:get_type() should be added to the API. But the function entity:get_movement_type() does not seem necessary at all (I think calling entity:get_movement():get_type() is a bit more natural).

**Comment by christopho** [comment]
Ok.

There is a way to implement it in pure Lua if you don't want to wait. The idea is to call getmetatable(movement) and see if it is the same as sol.main.get_metatable("straight_movement"), or sol.main.get_metatable("circle_movement"), etc.

**Comment by wrightmat** [comment]
Makes sense, I may use that in the mean time. Thanks!

And yeah, @Diarandor, I meant one or the other. I also prefer entity:get_movement():get_type()

**Comment by christopho** [comment]
It will be movement:get_type() yes. Remember that movements can also be used without entities, for example you can move a surface in your title screen.

**Comment by Diarandor** [comment]
An (untested) workaround for this could be to redefine the "sol.movement.create", so the type of the movement is saved in some variable, or you can just define the get_type() function there. You could write something similar to this in some script and load it on the main script:
```
-- Redefine "sol.movement.create" to save the movement type and define movement:get_type().
local old_movement_create = sol.movement.create
sol.movement.create = function(movement_type)
  local m = old_movement_create(movement_type)
  function m:get_type() return movement_type end
  return m
end
old_movement_create = nil -- Destroy this useless local variable.
```
